### PR TITLE
Add eth_sign

### LINKED ISSUE #766: Eth RPC for eth_sign
Implement the eth_sign RPC in web3signer.

https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sign

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
Add eth_sign RPC method under eth1 subcommand

## Fixed Issue(s)

Fixes #766
 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/internalresponse/EthSignResultProvider.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    final String identifier = params.get(0);
+    final Optional<ArtifactSigner> transactionSigner =
+        transactionSignerProvider.getSigner(identifier);
+    if (transactionSigner.isEmpty()) {
+      LOG.debug("Address ({}) does not match any available account", identifier);
+      throw new JsonRpcException(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT);
+    }
+    final Bytes ethMessage = getEthereumMessage(params.get(1));
+
+    return transactionSignerProvider.sign(normaliseIdentifier(identifier), ethMessage).get();
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/internalresponse/EthSignResultProvider.java:64]: Could simplify this to just use the transactionSignerProvider.sign and check the result for an empty to signify that it didn't match any account. Then also wouldn't need to add the getSigner on SignerForIdentifier.

```suggestion
        final Bytes ethMessage = getEthereumMessage(params.get(1));
    return transactionSignerProvider
        .sign(identifier, ethMessage)
        .orElseThrow(
            () -> {
              LOG.debug("Address ({}) does not match any available account", identifier);
              return new JsonRpcException(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT);
            });
```

Comment (gfukushima) [reply core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/internalresponse/EthSignResultProvider.java:64]: Do we need to remove the normalisation?

**Review by jframe — APPROVED** [review]
LGTM

**Comment by bgravenorst** [comment]
Removing doc-change-required label since this is documented here: https://docs.web3signer.consensys.net/Reference/API/json-rpc#eth_sign

**Pull request merged**